src/seedu/addressbook/parser/Parser.java

```diff
@@ -174,6 +177,59 @@ private Command prepareDelete(String args) {
         }
     }
 
+    /**
+     * Parses arguments in the context of the edit person command.
+     *
+     * @param args full command args string
+     * @return the prepared command
+     */
+    private Command prepareEdit(String args) {
+        if (args.split(" ").length <= 4) {
+            return new IncorrectCommand(String.format(MESSAGE_INVALID_COMMAND_FORMAT, seedu.addressbook.commands.EditCommand.MESSAGE_USAGE));
+        }
+
+        args = args.trim();
+
+        int i = args.indexOf(" ");
+        String index = args.substring(0, i);
+        String pArgs = args.substring(i);
+        final int targetIndex;
```

Comment:
The level of abstraction here is not consistent. Try to refactor this part of the code into separate methods to ensure better readability and coding practice.